components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -673,10 +552,23 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
     diag_rhoi = 0.
     rhorime_c = 400.
 
-    tmparr1 = (pres*1.e-5)**(rd*inv_cp)
+    call get_precip_fraction(its,ite,kts,kte,kbot,ktop,kdir,ast,qc,qr,qitot,precip_frac_method, &
+                cldm,icldm,lcldm,rcldm)
+    cmeiout = 0.
+    prain   = 0.
+    nevapr  = 0.
+    rflx    = 0.
+    sflx    = 0.
+
+    ! AaronDonahue added exner term to replace all instances of th(i,k)/t(i,k), since th(i,k) is updated but t(i,k) is not, and this was
+    ! causing energy conservation errors.
+    inv_exner = 1./exner        !inverse of Exner expression, used when converting potential temp to temp
+    tmparr1 = inv_exner !(pres*1.e-5)**(rd*inv_cp)
```

Comment:
What is the purpose of tmparr1 here? Could you just use inv_exner to improve readability?

Reply:
I agree that it would just be simpler to use inv_exner on the next line instead of having a tmparr1.  I left it in because I didn't know how many changes we wanted to make from the original P3.  I'm happy to make the change now, or we will probably have a code clean-up stage where we can address things like this.

Reply:
Ok, that sounds fine to me. Let's push this off to code clean-up.

Reply:
might as well make this small change now...

Reply:
Sounds good